### PR TITLE
feat(eslint-config): Add support for TypeScript files in ESLint and remove tslint-config

### PR DESCRIPTION
**Pull request by lipis**
BREAKING CHANGE: Add support for TypeScript files in ESLint and remove tslint-config

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -3,5 +3,23 @@
...
+  ],
+  "parserOptions": {
+    "project": "./tsconfig.json"
+  },
+  "plugins": ["jasmine"],
+  "rules": {
+    "id-length": "off",
+    "no-dupe-class-members": "off",
+    "no-magic-numbers": "off",
+    "valid-jsdoc": "off"
```
Comment (lipis) [inline .eslintrc.json]: in the upcoming PRs we could remove some of these overrides.

File: packages/api-client/src/account/AccountAPI.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ export class AccountAPI {
    * Retrieve all TURN server addresses and credentials.
    * Clients are expected to do a DNS lookup to resolve the IP addresses of the given hostnames
    *
-   * @param limit Limits the number of ICE-Candidates returned. [1..10]
```
Comment (ffflorian) [inline packages/api-client/src/account/AccountAPI.ts]: Please remove the dashes in TS files, see here why: https://github.com/wireapp/wire-webapp/pull/7904#discussion_r358145110

File: packages/core/src/demo/status-bot.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ if (!message) {
 
     const text = message || `I am posting from ${name} v${version} (Build #${process.env.TRAVIS_BUILD_NUMBER}). 🌞`;
     const payload = account.service.conversation.createText(text).build();
-    conversationIds.forEach(async conversationId => await account.service.conversation.send(conversationId, payload));
+    conversationIds.forEach(async conversationId => account.service.conversation.send(conversationId, payload));
```
Comment (ffflorian) [inline packages/core/src/demo/status-bot.js]: Why did you remove `await` here?

File: packages/cryptobox/src/demo/benchmark.js [hunk 4]
```diff
@@ -66,7 +66,7 @@ async function encryptBeforeDecrypt({alice, bob}, messageCount) {
   process.stdout.write(`Measuring decryption time for "${messageCount}" messages ... `);
   startTime = process.hrtime();
   await Promise.all(
-    encryptedMessages.map(async encryptedMessage => await bob.decrypt(createSessionId(alice), encryptedMessage)),
+    encryptedMessages.map(async encryptedMessage => bob.decrypt(createSessionId(alice), encryptedMessage)),
```
Comment (ffflorian) [inline packages/cryptobox/src/demo/benchmark.js]: Why did you remove `await` here?

File: packages/eslint-config/package.json [hunk 5]
```diff
@@ -1,20 +1,34 @@
...
     "eslint-plugin-react": ">=7",
-    "eslint-plugin-simple-import-sort": ">=4",
+    "eslint-plugin-react-hooks": ">=2",
     "eslint-plugin-sort-keys-fix": ">=1",
-    "prettier": ">=1.17"
+    "prettier": ">=1.19"
   },
   "repository": "https://github.com/wireapp/wire-web-packages/tree/master/packages/eslint-config",
-  "version": "1.3.0"
+  "version": "1.3.0-beta.22"
```
Comment (ffflorian) [inline packages/eslint-config/package.json]: Let's remove the beta tag

File: packages/proteus/src/main/util/MemoryUtil.ts [hunk 6]
```diff
@@ -29,6 +29,5 @@ export function zeroize(object: Uint8Array | ArrayBuffer | Record<string, any> |
       .map(key => object[key])
       .forEach(val => zeroize(val));
   } else {
-    return;
   }
```
Comment (ffflorian) [inline packages/proteus/src/main/util/MemoryUtil.ts]: We can remove this `else`.

File: packages/api-client/src/team/member/Permissions.ts [hunk 7]
```diff
@@ -17,7 +17,6 @@
  *
  */
 
-/* tslint:disable:object-literal-sort-keys */
 export enum Permissions {
```
Comment (ffflorian) [inline packages/api-client/src/team/member/Permissions.ts]: This file still has `tslint:enable`.

File: packages/api-client/src/account/AccountAPI.ts (hunk 2 above)
Comment (lipis) [reply packages/api-client/src/account/AccountAPI.ts]: They were added automatically, do you know which rule is that?

File: packages/core/src/demo/status-bot.js (hunk 3 above)
Comment (lipis) [reply packages/core/src/demo/status-bot.js]: Redundant use of `await` on a return

File: packages/eslint-config/package.json (hunk 5 above)
Comment (lipis) [reply packages/eslint-config/package.json]: Will the major version increased automatically or we have to do something? I already have a BREAKING CHANGE in the description

File: packages/core/src/demo/status-bot.js (hunk 3 above)
Comment (ffflorian) [reply packages/core/src/demo/status-bot.js]: `await` on a return isn't always redundant: https://jakearchibald.com/2017/await-vs-return-vs-return-await/

Comment (lipis) [reply packages/core/src/demo/status-bot.js]: But in this case it seems redundant, from the article:

> _Outside of try/catch blocks, return await is redundant. There's even an ESLint rule to detect it, but it allows it in try/catch._

Comment (lipis) [reply packages/core/src/demo/status-bot.js]: and according to ESLint..

Comment (ffflorian) [reply packages/core/src/demo/status-bot.js]: Ok! It's just a demo anyway :)
